From fd4d0676adc10cbcfb4032b506933a529d815baf Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 16 Jul 2018 14:04:02 +0800
Subject: [PATCH 63/73] rk3308-evb-dmic-pdm-v11-aarch32: Disable vad

Run arecord with 44.1K would cause crash, so disable vad for now.

Change-Id: Ibe0600e0b7190860cfb34ff42f3d2fd1de69e94e
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 .../boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts  | 40 +++++-----------------
 1 file changed, 9 insertions(+), 31 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts
index 7932d0a8545d..0f2416551b41 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts
@@ -11,17 +11,8 @@
 	model = "Rockchip RK3308 evb digital-pdm mic v11 board";
 	compatible = "rockchip,rk3308-evb-dmic-pdm-v11", "rockchip,rk3308";
 
-	pdm_i2s_dais: pdm-i2s-dais {
-		status = "okay";
-		compatible = "rockchip,rk3308-multi-dais", "rockchip,multi-dais";
-		dais = <&pdm_8ch>, <&i2s_8ch_2>;
-		capture,channel-mapping = <6 2>;
-		playback,channel-mapping = <0 2>;
-		bitclock-inversion = <1 0>;
-	};
-
 	pdm-mic-array {
-		status = "disabled";
+		status = "okay";
 		compatible = "simple-audio-card";
 		simple-audio-card,name = "rockchip,pdm-mic-array";
 		simple-audio-card,cpu {
@@ -33,27 +24,26 @@
 	};
 
 	vad-sound {
-		status = "okay";
+		status = "disabled";
 		compatible = "rockchip,multicodecs-card";
 		rockchip,card-name = "rockchip,rk3308-vad";
-		rockchip,cpu = <&pdm_i2s_dais>;
-		rockchip,codec = <&acodec>, <&vad>;
+		rockchip,cpu = <&pdm_8ch>;
+		rockchip,codec = <&dummy_codec>, <&vad>;
 	};
 };
 
 &acodec_sound {
-	status = "disabled";
+	status = "okay";
 };
 
 &dummy_codec {
-	status = "disabled";
+	status = "okay";
 	#sound-dai-cells = <0>;
 };
 
 &pdm_8ch {
 	status = "okay";
 	#sound-dai-cells = <0>;
-	rockchip,no-dmaengine;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pdm_m2_clk
 		     &pdm_m2_clkm
@@ -64,21 +54,9 @@
 };
 
 &vad {
-	status = "okay";
-	rockchip,audio-src = <&pdm_8ch>;
-	rockchip,det-channel = <0>;
+	status = "disabled";
+	rockchip,audio-src = <4>;
+	rockchip,det-channel = <2>;
 	rockchip,mode = <1>;
-	rockchip,buffer-time-ms = <200>;
-	#sound-dai-cells = <0>;
-};
-
-&i2s_8ch_2 {
-	status = "okay";
-	rockchip,no-dmaengine;
-	#sound-dai-cells = <0>;
-};
-
-&pdm_i2s_dais {
-	status = "okay";
 	#sound-dai-cells = <0>;
 };
-- 
2.11.0

